core/src/main/java/org/aerogear/mobile/core/ServiceModule.java

```diff
@@ -2,14 +2,25 @@
 
 import org.aerogear.mobile.core.configuration.ServiceConfiguration;
 
-
 public interface ServiceModule {
 
     /**
-     * This is the method called by mobile-core to startup modules.
-     * @param core an instance of mobile core
-     * @param configuration service configuration for this instance of a service module
-     * @throws BootstrapException if bootstrapping fails.
+     * Type/name used in the mobile-service.json
+     *
+     * @return
+     */
+    String type();
+
+    /**
+     * A method how create/configure the service
+     *
+     * @param serviceConfiguration
      */
-    void bootstrap(MobileCore core, ServiceConfiguration configuration);
+    void configure(ServiceConfiguration serviceConfiguration);
```

Comment:
Let's pass core here.  This way modules can reference modules they depend on by using MobileCore#getInstance($Module.class).  In getInstance modules are already lazy loaded.  The only catch is that circular dependencies will trigger a stack overflow exception.